fastembed/rerank/cross_encoder/onnx_text_model.py

```diff
@@ -40,23 +26,24 @@ def load_onnx_model(
         )
         self.tokenizer, _ = load_tokenizer(model_dir=model_dir)
         self.configure_tokenizer()
+        self.ONNX_INPUT_NAMES = [input_meta.name for input_meta in self.model.get_inputs()]
 
     def configure_tokenizer(self) -> None:
         """Configures the tokenizer to properly handle query and document pairs."""
-        cls_token_id = self.tokenizer.token_to_id("[CLS]")
-        sep_token_id = self.tokenizer.token_to_id("[SEP]")
-
-        if cls_token_id is not None and sep_token_id is not None:
-            self.tokenizer.post_processor = TemplateProcessing(
-                single="[CLS] $A [SEP]",
-                pair="[CLS] $A [SEP] $B:1 [SEP]:1",
-                special_tokens=[
-                    ("[CLS]", cls_token_id),
-                    ("[SEP]", sep_token_id),
-                ],
-            )
-        else:
-            print("Tokenizer does not have [CLS] or [SEP] tokens. Skipping post-processing setup.")
+        cls_token_id = self.tokenizer.token_to_id("[CLS]") or self.tokenizer.token_to_id("<s>")
+        sep_token_id = self.tokenizer.token_to_id("[SEP]") or self.tokenizer.token_to_id("</s>")
+
+        if cls_token_id is None or sep_token_id is None:
+            raise ValueError("Tokenizer does not have [CLS] or [SEP] tokens, or their equivalent.")
+
+        self.tokenizer.post_processor = TemplateProcessing(
+            single="[CLS] $A [SEP]",
+            pair="[CLS] $A [SEP] $B:1 [SEP]:1",
+            special_tokens=[
+                ("[CLS]", cls_token_id),
+                ("[SEP]", sep_token_id),
+            ],
+        )
```

Comment:
This scheme does not work for `BAAI/bge-reranker-base`

Compare it to the one obtained via transformers:

```
import torch
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("BAAI/bge-reranker-base")


pairs = [
    ["What is the capital of France?", "Paris is the capital of France."],
    ["What is the capital of France?", "Berlin is the capital of Germany."],
]
with torch.no_grad():
    inputs = tokenizer(pairs, padding=True, truncation=True, return_tensors='np', max_length=512)
    print(inputs)
```

```
# array([[     0,   4865,     83,     70,  10323,    111,   9942,     32,
#              2,      2,   7270,     83,     70,  10323,    111,   9942,
#              5,      2],
#        [     0,   4865,     83,     70,  10323,    111,   9942,     32,
#              2,      2,  10271,     83,     70,  10323,    111, 102126,
#              5,      2]]),
```

Ours:

```
# [[0, 4865, 83, 70, 10323, 111, 9942, 32, 2, 7270, 83, 70, 10323, 111, 9942, 5, 2], [0, 4865, 83, 70, 10323, 111, 9942, 32, 2, 10271, 83, 70, 10323, 111, 102126, 5, 2]
```

Take a look at `111, 9942, 32, 2, 7270, 83,`
Transformers output has 2 `2`